bentoml/server/api_server.py

```diff
@@ -536,52 +556,120 @@ def handle_ml_predict(self):
             'Authorization': 'Bearer ' + access_token,
             'Content-Type': 'application/ld+json'
         }
-        MLMODEL_UUID = self.ngsild_ml_model_id
+        
+        WEATHER_UUID = 'urn:ngsi-ld:WeatherObserved:Pegomas'
         URL_ENTITIES = self.ngsild_cb_url + '/ngsi-ld/v1/entities/'
+        URL_TEMPORAL = self.ngsild_cb_url + '/ngsi-ld/v1/temporal/entities/'
         AT_CONTEXT = [ self.ngsild_at_context ]
-        ATTRIBUTE_INPUT_DATA = self.ngsild_ml_model_input
-        ATTRIBUTE_OUTPUT_DATA = self.ngsild_ml_model_output
+        TARGET_ENTITY = self.ngsild_ml_model_target_entity
 
         # Get the POST data
+        # We actually don't care about it, as we actually need to get
+        # data from different sources, i.e. not only precipitation but
+        # temperature as well.
+        # This notification just acts as atrigger
         input_data_notification = request.get_json()
-
-        input_entity = input_data_notification['data'][0]['id']
-        input_data = input_data_notification['data'][0][ATTRIBUTE_INPUT_DATA]['value']
-        logger.info('input_data received from notification: %s', input_data)
-
-        # reshaping input data into a 2D array
-        input_data = np.array([input_data]).reshape(-1,1)
+        logger.info('input_data received from notification: %s', input_data_notification)
+
+        # To predict next three months of precipitation/temperature
+        # We get the current date - 90 days, then perform get temporal
+        # with 30 days aggregation to retrieve THREE values of precipitation
+        # and temperature from WeatherObserved Pegomas
+        # The ML model being used here predict the (precipitation, temperature) 
+        # of the next month from the three previous month.
+        _90DaysBefore = datetime.now(timezone.utc) - timedelta(days=90)
+        _90DaysBeforeStr = _90DaysBefore.strftime("%Y-%m-%dT%H:%M:%SZ")
+        
+        # precipitation (Aggregation SUM)
+        query = '?attrs=https%3A%2F%2Furi.fiware.org%2Fns%2Fdata-models%23precipitation&timerel=after&time=' \
```

Comment:
this query (and the other one) could / should be more dynamic / configurable

Reply:
As I wrote earlier, a lot of part should be more configurable, we just need to take some time to think about different use cases and try to come up with a generic behaviour.
I'm so far not in a position to do so, trying to have something running asap.